core/src/main/java/org/infinispan/statetransfer/OutboundTransferTask.java

```diff
@@ -285,8 +286,8 @@ public String toString() {
       return "OutboundTransferTask{" +
             "topologyId=" + topologyId +
             ", destination=" + destination +
-            ", segments=" + segments +
-            ", stateTransferChunkSize=" + stateTransferChunkSize +
+            ", segments=" + new IntSet(segments) +
```

Comment:
hmmm... on other hand, you will need to order the ```Set```. I'm not sure if it can be done.